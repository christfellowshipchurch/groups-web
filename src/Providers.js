import React from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';

import { defaultTheme } from './ui-kit/theme';

const Providers = ({ children }) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
);

Providers.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Providers;
