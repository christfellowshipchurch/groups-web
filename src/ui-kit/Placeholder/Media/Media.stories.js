import React from 'react';
import { storiesOf } from '@storybook/react';

import 'react-placeholder/lib/reactPlaceholder.css';

import Media from '.';

storiesOf('ui-kit/Placeholder/Media', module)
  .addDecorator((story) => (
    <div style={{ width: '50vw', height: '50vw' }}>{story()}</div>
  ))
  .add('default', () => <Media />)
  .add('width', () => <Media size={'50%'} />)
  .add('showLoadingAnimation', () => <Media showLoadingAnimation />);
