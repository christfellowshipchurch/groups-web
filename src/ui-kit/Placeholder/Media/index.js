import React from 'react';
import styled from 'styled-components';
import ReactPlaceholder from 'react-placeholder';
import PropTypes from 'prop-types';

import { withOverrides } from '../../theme';
import { getIsLoading } from '../../isLoading';

const StyledMedia = styled.mark.attrs(
  withOverrides('uiKit.Placeholder.Media.StyledMedia', () => ({
    // `custom-placeholder` class enables the `showLoadingAnimation` prop to work
    className: 'custom-placeholder',
  }))
)`
  background: ${({ theme }) => theme.colors.background.inactive};
  border-radius: ${({ borderRadius, theme }) =>
    borderRadius || theme.sizing.baseBorderRadius};
  display: block;
  width: ${({ size }) => size};
  padding: ${({ size }) => `${size} 0 0 0`};

  /* Styles passed via the style prop and/or overrides */
  ${({ $style }) => $style}
`;

const Media = getIsLoading(({ isLoading, size, style, ...props }) => (
  <ReactPlaceholder
    customPlaceholder={<StyledMedia size={size} $style={style} />}
    ready={!isLoading}
    {...props}
  />
));

Media.defaultProps = {
  isLoading: true,
  size: '100%',
};

Media.propTypes = {
  isLoading: PropTypes.bool,
  size: PropTypes.string,
  style: PropTypes.any, // eslint-disable-line react/forbid-prop-types
};

Media.displayName = 'ui-kit.Placeholder.Media';

export default styled(Media).attrs(
  withOverrides('uiKit.Placeholder.StyledMedia')
)``;
