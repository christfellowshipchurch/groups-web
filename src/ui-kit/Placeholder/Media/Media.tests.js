import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Providers from '../../../Providers';

import Media from '.';

describe('The Media placeholder', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <Media />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a custom size', () => {
    const tree = renderer.create(
      <Providers>
        <Media size="50%" />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
