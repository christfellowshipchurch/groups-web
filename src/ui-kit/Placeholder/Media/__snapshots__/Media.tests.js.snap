// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Media placeholder accepts custom styles 1`] = `
.c0 {
  background: #A5A5A5;
  border-radius: 5px;
  display: block;
  width: 100%;
  padding: 100% 0 0 0;
  background: salmon;
}

<mark
  className="c0 custom-placeholder"
  size="100%"
/>
`;

exports[`The Media placeholder should render 1`] = `
.c0 {
  background: #A5A5A5;
  border-radius: 5px;
  display: block;
  width: 100%;
  padding: 100% 0 0 0;
}

<mark
  className="c0 custom-placeholder"
  size="100%"
/>
`;

exports[`The Media placeholder should render a custom size 1`] = `
.c0 {
  background: #A5A5A5;
  border-radius: 5px;
  display: block;
  width: 50%;
  padding: 50% 0 0 0;
}

<mark
  className="c0 custom-placeholder"
  size="50%"
/>
`;

exports[`The Media placeholder should render children when isLoading = false 1`] = `"Rendered child"`;

exports[`The Media placeholder should render showLoadingAnimation 1`] = `
.c0 {
  background: #A5A5A5;
  border-radius: 5px;
  display: block;
  width: 100%;
  padding: 100% 0 0 0;
}

<mark
  className="c0 show-loading-animation custom-placeholder"
  size="100%"
/>
`;
